src/Knp/Component/Pager/Paginator.php

```diff
@@ -88,16 +89,15 @@ public function setDefaultPaginatorOptions(array $options): void
      *     boolean $distinct - default true for distinction of results
      *     string $alias - pagination alias, default none
      *     array $whitelist - sortable whitelist for target fields being paginated
-     * @throws \LogicException
+     * @throws PagerBadArgumentException
      * @return PaginationInterface
      */
     public function paginate($target, int $page = 1, int $limit = 10, array $options = []): PaginationInterface
     {
-        $limit = intval(abs($limit));
-        if (!$limit) {
-            throw new \LogicException("Invalid item per page number, must be a positive number");
+        if ( $limit<=0 or $page<=0) {
```

Comment:
Wrong spacing here: you should omit spaces after/before parentehes and add spaces after/before controls (like `<=`